kernel/thread.c

```diff
@@ -995,3 +1003,64 @@ static inline k_ticks_t z_vrfy_k_thread_timeout_expires_ticks(
 }
 #include <syscalls/k_thread_timeout_expires_ticks_mrsh.c>
 #endif
+
+#ifdef CONFIG_THREAD_RUNTIME_STATS
+void z_thread_mark_switched_in(void)
+{
+	struct k_thread *thread;
+
+	thread = k_current_get();
+	thread->rt_stats.last_switched_in = k_cycle_get_32();
```

Comment:
shouldnt you be getting the timing information using the timing APIs?

Reply:
Another commit adds the kconfig to use timing functions. This has the flexibility to not use timing functions as the timer used by timing functions may be reserved for something else in the app.